tests/Functional/Driver/IBMDB2/StatementTest.php

```diff
@@ -38,4 +38,18 @@ public function testExecutionErrorsAreNotSuppressed(): void
         $this->expectException(StatementError::class);
         $stmt->execute([[]]);
     }
+
+    public function testWarningDoNotBreakFetchAll(): void
+    {
+        $driverConnection = $this->connection->getWrappedConnection();
+
+        $stmt = $driverConnection->prepare('SELECT current time AS time FROM sysibm.sysdummy1');
+
+        // prevent the PHPUnit error handler from handling the errors that db2_execute() may trigger
+        $this->iniSet('error_reporting', (string) E_ALL);
+
+        $result = $stmt->execute([[]])->fetchAllAssociative();
```

Comment:
It is known that `SELECT ... FROM sysibm.sysdummy1` will always return a single row. Why do you use `fetchAllAssociative()`?

Reply:
The error happens when trying to retrieve a line from a collection that does not exist (to know that the collection is finished). Using fetchAssociative there's no problem since a line exist.